Request to adding base Java classes architecture

Alvahaze opened the pull request:
- Class structure according to UML was created  752f72a
- Little fix in UML  802706b
- Some fixes in base Java classes architecture  7db992f
- Small fix in UML diagram  dba3633

pisarevski committed:
Class structure according to UML was created

Packages:
- ammunition
- events
- locations
- parts
- ships
- trade

Classes:
- Storage
- Universe
- Ammo
- Gun
- Magazine
- Railgun
- Turret
- Event
- Asteroid
- Location
- OpenSpace
- Planet
- Engine
- Hull
- Starship
- PirateStarship
- Traider
- Mechanic
- AverageTrader

Interfaces:
- ILandable
- IStorageable
- IAttackable

pisarevski committed:
Little fix in UML

- Little fix in UML, in class Starship

ViRGiL175 commented on src/main/java/rape/brutal/gitstarship/Storage.java:
```diff
@@ -0,0 +1,24 @@
+ * Created by haze on 22.03.2017.
+ */
+public class Storage {
+        private HashMap<String, IStorageable> items;
```
Bad naming

Alvahaze replied:
All fixed!

ViRGiL175 reviewed (changes requested):
- Shitty code arranging
- Access modifiers errors
- Dirty interfaces
- Getters and setters needed

ViRGiL175 commented:
Reviewed.

pisarevski committed:
Some fixes in base Java classes architecture

- Rename items to itemsHashMap in class Storage
- Add getters and setters everywhere
- Code refactored
- Add modifer public to class AverageTrader
- Removed modifer public in interfaces

pisarevski committed:
Small fix in UML diagram

- Rename items to itemsHashMap in class Storage

ViRGiL175 commented on src/main/java/rape/brutal/gitstarship/locations/Location.java:
```diff
@@ -0,0 +1,33 @@
+/**
+ * Created by haze on 22.03.2017.
+ */
+public abstract class Location {
```
Line gap was lost.

ViRGiL175 commented:
Reviewed.
